src/components/manifold-subscription-details/store.ts

```diff
@@ -0,0 +1,41 @@
+import { createStore } from '@stencil/store';
+import { Connection } from '@manifoldco/manifold-init-types/types/v0';
+import { GraphqlError } from '@manifoldco/manifold-init-types/types/v0/graphqlFetch';
+import { ProductPlansQuery, ProductPlansQueryVariables } from '../../types/graphql';
+import productPlansQuery from './product-plans.graphql';
+
+interface Store {
+  connection?: Connection;
+  productId?: string;
+  isLoading: boolean;
+  plans?: ProductPlansQuery['product']['plans']['edges'];
+  errors?: GraphqlError[];
+}
+
+const { state, onChange } = createStore<Store>({
+  isLoading: false,
+});
+
+onChange('productId', async (productId: string) => {
+  if (!state.connection?.graphqlFetch) {
+    return;
+  }
+
+  state.isLoading = true;
+
+  const variables: ProductPlansQueryVariables = { productId };
+  const { data, errors } = await state.connection.graphqlFetch<ProductPlansQuery>({
+    query: productPlansQuery,
+    variables,
+  });
+
+  state.errors = errors;
+
+  if (data) {
+    state.plans = data.product.plans.edges;
+  }
+
+  state.isLoading = false;
+});
+
+export default state;
```

Comment:
Probably going to end up creating multiple stores. I'll be making one probably for cost calculations as well. Also, Once we start using this for the subscription create component, we'll want to convert this to a store factory so we can create multiple independent stores, just in case those components exist on the same page for some reason.

Reply:
+1 on multiple stores. Different concerns should get isolated state.